lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart

```diff
@@ -67,39 +78,96 @@ class _UpdatePhoneNoBottomSheetBody
-                                          HSColor.neutral10),
-                                      foregroundColor:
-                                          MaterialStatePropertyAll<Color>(
-                                              HSColor.black),
-                                      textStyle:
-                                          MaterialStatePropertyAll<TextStyle>(
+                          ValueListenableBuilder<PhoneNumberErrorType?>(
+                            valueListenable: _phoneNumberError,
+                            builder: (context, value, child) {
+                              return Row(
+                                crossAxisAlignment: CrossAxisAlignment.start,
+                                children: [
+                                  PrimaryButton(
+                                      label: HatSpaceStrings
+                                          .current.countryCode,
+                                      style: const ButtonStyle(
+                                          alignment: Alignment.centerLeft,
+                                          backgroundColor:
+                                              MaterialStatePropertyAll(
+                                                  HSColor.neutral10),
+                                          textStyle: MaterialStatePropertyAll<
+                                                  TextStyle>(
                                               TextStyle(
-                                        fontSize: FontStyleGuide.fontSize14,
-                                        fontWeight: FontWeight.w400,
-                                      )))),
-                              const SizedBox(width: HsDimens.spacing24),
-                              Expanded(
-                                  child: HatSpaceInputText(
-                                label: '',
-                                onChanged: (value) {
-                                  // TODO: Adding phone number logic
-                                },
-                                placeholder: HatSpaceStrings
-                                    .current.updatePhonePlaceHolder,
-                              ))
-                            ],
-                          )
+                                                  fontSize:
+                                                      FontStyleGuide.fontSize14,
+                                                  fontWeight: FontWeight.w400,
+                                                  color: HSColor.neutral9)))),
+                                  const SizedBox(width: HsDimens.spacing24),
+                                  Expanded(
+                                      child: TextFormField(
+                                    key: const Key('phoneNo'),
+                                    onChanged: (value) {
+                                      setState(() {
```

Comment:
Please use bloc, avoid using setState when bloc is in used

Reply:
Fixed with cubit